
<!--
  Manipulator specific version of 'robot_interface_streaming.launch'.

  Defaults provided for sia5d:
   - 7 joints

  Usage:
    robot_interface_streaming_sia5d.launch robot_ip:=<value> controller:=<fs100|dx100>
-->
<launch>
	<arg name="robot_ip" />
	
  <!-- controller: Controller name (fs100 or dx100) -->
  <arg name="controller" default="fs100"/>

  <arg name="in_joint_command_topic" default="simple_joint_command"/>

  <arg name="out_joint_command_topic" default="joint_command"/>

  <arg name="joint_state_topic" default="joint_states"/>

  <arg name="get_joints_service" default="getJoints"/>

	<rosparam command="load" file="$(find motoman_sia5d_support)/config/joint_names_sia5d.yaml" />

	<include file="$(find motoman_driver)/launch/robot_interface_streaming_$(arg controller).launch">
		<arg name="robot_ip"   value="$(arg robot_ip)" />
	</include>

  <node name="motoman_interface"  pkg="motoman_interface" type="motoman_interface" output="screen">

      <param name="in_joint_command_topic" type="string" value="$(arg in_joint_command_topic)"/>

      <param name="out_joint_command_topic" type="string" value="$(arg out_joint_command_topic)"/>

      <param name="joint_state_topic" type="string" value="$(arg joint_state_topic)"/>

      <param name="get_joints_service" type="string" value="$(arg get_joints_service)"/>

</node>

</launch>
